crates/rpc/src/v02/types.rs

```diff
@@ -423,7 +423,7 @@ pub mod request {
             ];
             starknet_gateway_types::transaction_hash::compute_v3_txn_hash(
                 b"declare",
-                TransactionVersion::THREE,
+                self.version,
```

Comment:
It would have been nice if the transaction variant did not have a version, but instead only a query flag. Then we could have done
```
TransactionVersion::THREE.with_query_flag(self.query_flag)
```